docs/src/main/asciidoc/extension-authors-guide.adoc

```diff
@@ -567,3 +570,483 @@ memory in a runtime class loader. To have these classes written out to disk for
 ----
 ./mvnw clean install -Dquarkus.debug.generated-classes-dir=./target/app-generated-classes
 ----
+
+
+== Sample Test Extension
+We have an extension that is used to test for regressions in the extension processing. It is located in {quarkus-tree-url}/core/test-extension directory. In this section we touch on some of the tasks an extension
+author will typically need to perform using the test-extension code to illustrate how the task could be done.
+
+=== Features and Capabilities
+When you startup a {project-name} instance, you will see a line like the one highlighted in this example:
+
+.Example Startup Lines
+[source, bash]
+----
+2019-03-22 14:02:37,884 INFO  [io.quarkus] (main) Quarkus 999-SNAPSHOT started in 0.061s. 
+2019-03-22 14:02:37,884 INFO  [io.quarkus] (main) Installed features: [cdi, test-extension] <1>
+----
+<1> A list of features installed in the runtime image
+
+The features listed reflect the types of extensions that are installed. An extension declares its features using a <<Build Step Processors>> method that produces a `FeatureBuildItem` like this TestProcessor#featureBuildItem() method example:
+
+
+.TestProcessor#featureBuildItem()
+[source,java]
+----
+    /**
+     * Register a extension capability and feature
+     * 
+     * @return test-extension feature build item
+     */
+    @BuildStep(providesCapabilities = "io.quarkus.test-extension")
+    FeatureBuildItem featureBuildItem() {
+        return new FeatureBuildItem("test-extension");
+    }
+----
+
+The feature name can be used to select the extension when creating a project using the {project-name} maven plugin as shown in this example taken from the Writing JSON REST Services guide where the "resteasy-jsonb" feature is referenced:
```

Comment:
AFAIK, that is not true. The feature name is a totally different beast and is only used for display.

Reply:
I think so too - names and labels used by CLI are currently hard-coded in https://github.com/quarkusio/quarkus/blob/master/devtools/common/src/main/filtered/extensions.json

Reply:
@starksm64 could you address this one? The statement is false AFAIK. The feature is just there to be displayed in the feature list at startup.

Reply:
Ok, right. I have corrected that section to indicate that the feature name should match a label in the extensions.json entry so that there is a correspondence between the displayed name and the label which is actually what is used to select the extension.